File: 14_longestCommonPrefix/sol.cpp
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.size() == 0) {
            return "";
        }

        string result = "";

        for (int i = 0; i < strs[0].length(); i++) {
            char char_to_check = strs[0][i];

            for (const string& str : strs) {
                if (i >= str.length() || str[i] != char_to_check) {
                    return result;
                }
            }
            result += char_to_check;
        }

        return result;
    }
};

File: 14_longestCommonPrefix/sol.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        
        result = ""
        
        for i in range(len(strs[0])):
            char_to_check = strs[0][i]
            
            for string in strs:
                if i >= len(string) or string[i] != char_to_check:
                    return result
            
            result += char_to_check
        
        return result